Allow setting CA bundle from environment variable

The patch is specific to invocation from within yocto. In other
invocation scenarios, in the rare circulstance that we catually need
to overrride the trusted CA bundle location, this can be done with the
AWS_CA_BUNDLE variable. Therefore, it is not upstream appropriate.

Upstream-Status: Inappropriate [oe specific]
Signed-off-by: Truls Asheim <truls@squaremind.io>



diff --git a/util.cpp b/util.cpp
index 890770e..13d9b82 100644
--- a/util.cpp
+++ b/util.cpp
@@ -3,6 +3,7 @@
 #include <stdarg.h>
 #include <stdio.h>
 #include "util.h"
+#include "debug.h"
 
 #include <aws/core/Aws.h>
 #include <aws/core/client/ClientConfiguration.h>
@@ -18,6 +19,12 @@ Aws::Client::ClientConfiguration create_aws_config(std::string region) {
         awsConfig.region = region;
     }
 
+    const char *config_cafile_env = getenv("AWS_KMS_PKCS11_CAFILE");
+    if (config_cafile_env != NULL) {
+        debug("Using CA file %s", config_cafile_env);
+        awsConfig.caFile = config_cafile_env;
+    }
+
     const char* endpoint = std::getenv("AWS_ENDPOINT_URL");
     if (endpoint != nullptr) {
         const char* prefix = "https";
